From 0525f97f325e52692d6e5aad482704c38b599fa0 Mon Sep 17 00:00:00 2001
From: Phil Sutter <psutter@redhat.com>
Date: Mon, 2 Nov 2015 13:00:28 +0100
Subject: [PATCH] ss: return -1 if an unrecognized option was given

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1265240

commit f73105ab42795422f131fb711712e3c860df982f
Author: Phil Sutter <phil@nwl.cc>
Date:   Thu Oct 15 21:01:16 2015 +0200

    ss: return -1 if an unrecognized option was given

    When getopt_long encounters an option which has not been registered, it
    returns '?'. React upon that and call usage() instead of help() so ss
    returns with a non-zero exit status.

    Signed-off-by: Phil Sutter <phil@nwl.cc>

Signed-off-by: Phil Sutter <psutter@redhat.com>
---
 misc/ss.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/misc/ss.c b/misc/ss.c
index eecd537..9a6136e 100644
--- a/misc/ss.c
+++ b/misc/ss.c
@@ -2607,8 +2607,8 @@ int main(int argc, char *argv[])
 			printf("ss utility, iproute2-ss%s\n", SNAPSHOT);
 			exit(0);
 		case 'h':
-		case '?':
 			help();
+		case '?':
 		default:
 			usage();
 		}
-- 
1.8.3.1

